src/Microsoft.AspNet.Mvc.Core/ActionResults/RedirectToRouteResult.cs

```diff
@@ -45,7 +45,7 @@ public override void ExecuteResult([NotNull] ActionContext context)
         {
             var urlHelper = GetUrlHelper(context);
 
-            var destinationUrl = urlHelper.RouteUrl(RouteValues);
+            var destinationUrl = urlHelper.RouteUrl(RouteName, RouteValues);
```

Comment:
Thought the functional tests should be sufficient but yeah there are unit tests for this class...added a new test now...